Sources/iron/math/Quat.hx

```diff
@@ -255,6 +257,150 @@ class Quat {
 		return this;
 	}
 
+
+	/**
+	  Convert this quaternion to an Euler of arbitrary order.
+	  @param	the order of the euler to obtain
+			(in blender order, opposite from mathematical order)
+			can be "XYZ", "XZY", "YXZ", "YZX", "ZXY", or "ZYX".
+	  @return	A new YZX Euler that represents the same rotation as this
+				quaternion.
+	**/
+	// this method use matrices as a middle ground
+	// (and is copied from blender's internal code in mathutils)
+	// note: there are two possible eulers for the same rotation, blender defines the 'best' as the one with the smallest sum of absolute components
+	//	should we actually make that choice, or is just getting one of them randomly good?
+	// note2: it seems that this engine transforms a vector by using vector×matrix instead of matrix×vector, meaning that the outer transformations are on the RIGHT.
+	//	(…Except for quaternions, where the outer quaternions are on the LEFT.)
+	//	anywho, the way the elements of the matrix are ordered makes sense (first digit-> row ID, second digit->column ID) in this system.
+	public inline function toEulerOrdered(p:String): Vec4{
+		// normalize quat ?
+		
+		var q0:Float = SQRT2 * this.w;
```

Comment:
Is `Float` instead of `FastFloat` intentional here?

Reply:
er… no.
I'll fix this.

Reply:
Plus, it was useless, because Mat3 would convert those to FastFloats.
Anyways, it's fixed now.